astropy/coordinates/transformations.py

```diff
@@ -347,23 +347,41 @@ def get_transform(self, fromsys, tosys):
             self._composite_cache[fttuple] = comptrans
         return self._composite_cache[fttuple]
 
-    def lookup_name(self, name):
+    def lookup_name(self, name, suppress_error=True):
         """
         Tries to locate the coordinate class with the provided alias.
 
         Parameters
         ----------
         name : str
             The alias to look up.
+        suppress_error : bool (default value is False)
+            True : doesn't raise exception if 'name' isn't found. (returns None)
+            False : raises exception if 'name' isn't found.
 
         Returns
         -------
         coordcls
+        if suppress_error == True:
             The coordinate class corresponding to the ``name`` or `None` if
             no such class exists.
-        """
+        if suppress_error == False:
+            The coordinate class corresponding to the ``name`` or raises KeyError
+            if no such class exists.
 
-        return self._cached_names.get(name, None)
+        Raises
+        -------
+        KeyError
+        If the coordinate class corresponding to the ``name`` does not exists
+        and suppress_error is False.
+        """
+        if suppress_error:
+            return self._cached_names.get(name, None)
+        else:
+            if name in self._cached_names:
+                return self._cached_names[name]
+            else:
+                raise KeyError("name not found")
```

Comment:
You don't need to raise it explicitly, you can simply use

```
...
else:
    return self._cached_names[name]
```
That will raise the KeyError automatically if the `name` is not inside the `_cached_names`.